iosHyperskillApp/iosHyperskillApp/Sources/Modules/StudyPlan/Views/StudyPlanSectionItemView.swift

```diff
@@ -0,0 +1,182 @@
+import shared
+import SwiftUI
+
+extension StudyPlanSectionItemView {
+    struct Appearance {
+        let arrowIconSize: CGFloat = 32
+
+        let rightDetailIconWidthHeight: CGFloat = 12
+
+        let backgroundProgressColor = Color(ColorPalette.overlayGreenAlpha7)
+
+        let progressBadgeForegroundColor = Color(ColorPalette.secondary)
+        let progressBadgeBackgroundColor = Color(ColorPalette.overlayGreenAlpha12)
+        let progressBadgeCornerRadius: CGFloat = 4
+        let progressBadgeInsets = LayoutInsets(horizontal: 8, vertical: 4)
+
+        var buttonStyle = OutlineButtonStyle(
+            borderColor: .border,
+            alignment: .leading,
+            paddingEdgeSet: [],
+            backgroundColor: Color(ColorPalette.surface)
+        )
+    }
+}
+
+struct StudyPlanSectionItemView: View {
+    private(set) var appearance = Appearance()
+
+    let item: StudyPlanWidgetViewStateSectionItem
+    let onActivityTap: () -> Void
+
+    var body: some View {
+        Button(action: onActivityTap) {
+            HStack(alignment: .top) {
+                VStack(alignment: .leading, spacing: LayoutInsets.defaultInset) {
+                    Text(item.title)
+                        .font(.body)
+                        .foregroundColor(item.isClickable ? .primaryText : .secondaryText)
+                        .padding(.trailing)
+
+                    if let formattedProgress = item.formattedProgress {
+                        Text(formattedProgress)
+                            .font(.caption)
+                            .foregroundColor(appearance.progressBadgeForegroundColor)
+                            .padding(appearance.progressBadgeInsets.edgeInsets)
+                            .background(appearance.progressBadgeBackgroundColor)
+                            .cornerRadius(appearance.progressBadgeCornerRadius)
```

Comment:
If you feel that source files gets too large you can extract some views into their own files. For example, I think it's OK to extract progress badge into own SwiftUI view file.